src/fsspec_xrootd/xrootd.py

```diff
@@ -219,6 +321,163 @@ def ls(self, path: str, detail: bool = True, **kwargs: Any) -> list[Any]:
+                _myFile.read,
+                start,
+                end - start,
+                self.timeout,
+            )
+            if not status.ok:
+                raise OSError(f"Bytes failed to read from open file: {status.message}")
+            return data
+        except Exception as exc:
+            status, _n = await _async_wrap(
+                _myFile.close,
+                self.timeout,
+            )
+            raise OSError(f"Something went wrong in _cat_file(): {exc}")
+
+    async def _get_max_chunk_info(self, file: Any) -> tuple[int, int]:
+        data_server = file.get_property("DataServer")
+        if data_server not in XRootDFileSystem.cat_chunk_cache.keys():
+            status, result = await _async_wrap(
+                self._myclient.query, QueryCode.CONFIG, "readv_iov_max readv_ior_max"
+            )
+            if not status.ok:
+                raise OSError(
+                    f"Server query for vector read info failed: {status.message}"
+                )
+            max_num_chunks, max_chunk_size = map(int, result.split(b"\n", 1))
+            XRootDFileSystem.cat_chunk_cache.update(
+                {
+                    data_server: {
+                        "max_num_chunks": max_num_chunks,
+                        "max_chunk_size": max_chunk_size,
+                    }
+                }
+            )
+        return (
+            int(XRootDFileSystem.cat_chunk_cache[data_server]["max_num_chunks"]),
+            int(XRootDFileSystem.cat_chunk_cache[data_server]["max_chunk_size"]),
+        )
+
+    async def _cat_vector_read(
+        self,
+        path: str,
+        chunks: list[tuple[int, int]],
+        batch_partition: int,
+    ) -> dict[str, list[bytes]]:
+        try:
+            _myFile = client.File()
+            status, _n = await _async_wrap(
+                _myFile.open,
+                self.protocol + "://" + self.storage_options["hostid"] + "/" + path,
+                OpenFlags.READ,
+                self.timeout,
+            )
+            if not status.ok:
+                raise OSError(f"File did not open properly: {status.message}")
+
+            max_num_chunks, max_chunk_size = await self._get_max_chunk_info(_myFile)
+            vectors = _make_vectors(chunks, max_num_chunks, max_chunk_size)
+
+            coros = [_async_wrap(_myFile.vector_read, v, self.timeout) for v in vectors]
+
+            results = await _run_coros_in_chunks(
+                coros, batch_size=batch_partition, nofiles=True
+            )
+
+            deets = []
+            for result in results:
+                if not result[0].ok:
+                    raise OSError(
+                        f"File did not vector_read properly: {status.message}"
+                    )
+                for r in result[1]:
+                    deets.append(r.buffer)
+
+        except Exception as exc:
+            raise OSError(f"Something went wrong in _cat_vector_read(): {exc}")
```

Comment:
we don't need to catch and re-raise the exception, we can just let it raise naturally

Reply:
ok